.github/actions/composite/updateProtectedBranch/action.yml

```diff
@@ -0,0 +1,140 @@
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Validate target branch
+      if: ${{ !contains(fromJSON('["main", "staging", "production"]'), github.event.inputs.TARGET_BRANCH) }}
+      shell: bash
+      run: |
+        echo "Target branch must be one of ['main', 'staging', 'production]"
+        exit 1
+
+    # If updating main, SOURCE_BRANCH must not be empty
+    - name: Validate source branch
+      if: github.event.inputs.TARGET_BRANCH == 'main' && github.event.inputs.SOURCE_BRANCH == ''
+      shell: bash
+      run: |
+        echo "Cannot update main branch without specifying a source branch"
+        exit 1
+
+    # If updating staging, the source branch will always be main
+    # If updating production, the source branch will always be staging
+    - name: Set source branch
+      shell: bash
+      run: |
+        if [[ ${{ github.event.inputs.TARGET_BRANCH }} == 'staging' ]]; then
+          echo "SOURCE_BRANCH=main" >> "$GITHUB_ENV"
+        elif [[ ${{ github.event.inputs.TARGET_BRANCH }} == 'production' ]]; then
+          echo "SOURCE_BRANCH=staging" >> "$GITHUB_ENV"
+        else
+          echo "SOURCE_BRANCH=${{ github.event.inputs.SOURCE_BRANCH }}" >> "$GITHUB_ENV"
+        fi
+
+    - uses: Expensify/App/.github/actions/composite/setupGitForOSBotify@main
+      with:
+        GPG_PASSPHRASE: ${{ inputs.GPG_PASSPHRASE }}
+
+    - name: Checkout source branch
+      shell: bash
+      run: git checkout ${{ env.SOURCE_BRANCH }}
+
+    - name: Set New Version
+      shell: bash
+      run: echo "NEW_VERSION=$(npm run print-version --silent)" >> "$GITHUB_ENV"
+
+    - name: Create temporary branch to resolve conflicts
+      if: ${{ contains(fromJSON('["staging", "production"]'), github.event.inputs.TARGET_BRANCH) }}
+      shell: bash
+      run: |
+        git checkout ${{ github.event.inputs.TARGET_BRANCH }}
+        BRANCH_NAME=update-${{ github.event.inputs.TARGET_BRANCH }}-from-${{ env.SOURCE_BRANCH }}
+        git checkout -b "$BRANCH_NAME"
+        git merge -Xtheirs ${{ env.SOURCE_BRANCH }}
+        git push --set-upstream origin "$BRANCH_NAME"
+
+    - name: Create Pull Request
+      id: createPullRequest
+      shell: bash
+      run: |
+        gh pr create \
+          --title "Update version to ${{ env.NEW_VERSION }} on ${{ github.event.inputs.TARGET_BRANCH }}" \
+          --body "Update version to ${{ env.NEW_VERSION }}" \
+          --label "automerge" \
+          --base ${{ github.event.inputs.TARGET_BRANCH }}
+        sleep 5
+        echo "::set-output name=PR_NUMBER::$(gh pr view --json 'number' --jq '.number')"
+      env:
+        GITHUB_TOKEN: ${{ inputs.OS_BOTIFY_TOKEN }}
+
+    - name: Check changed files
+      if: ${{ github.event.inputs.TARGET_BRANCH == 'main' }}
+      id: changedFiles
+      # Version: 3.3.0
+      uses: umani/changed-files@1d252c611c64289d35243fc37ece7323ea5e93e1
+      with:
+        repo-token: ${{ github.token }}
```

Comment:
Note: I did [an experiment](https://github.com/Andrew-Test-Org/Public-Test-Repo/pull/302#issuecomment-1171732293) and confirmed that, even though secrets are not directly available in an action, the full github context, including `github.token` _is_ available.